tmt/steps/provision/__init__.py

```diff
@@ -923,6 +923,17 @@ def setup(self) -> None:
 
         Setup the guest after it has been started. It is called after :py:meth:`Guest.start`.
         """
+        from tmt.steps.execute.internal import effective_pidfile_root
+        sudo = 'sudo' if not self.facts.is_superuser else ''
+        pid_directory = effective_pidfile_root()
+        self.execute(ShellScript(
+            f"""
+            if [ ! -d {pid_directory} ]; then \
```

Comment:
perhaps the test for pid_directory could be done before the shell?
```
if not pid_directory.exists() and not pid_directory.is_dir:
    self.execute(ShellScript(
    ...
```

Reply:
We cannot use `pid_directory.exists()`, which runs on tmt runner, but we need to test the existence on the guest.